sopel/irc/__init__.py

```diff
@@ -301,6 +295,17 @@ def on_error(self):
         self.last_error_timestamp = datetime.now()
         self.error_count = self.error_count + 1
 
+    def change_current_nick(self, new_nick):
+        """Change the current nick without configuration modification.
+
+        :param str new_nick: New nick to be used by the bot.
+        """
+        self._nick = tools.Identifier(new_nick)
```

Comment:
Updating the bot's internal state before even sending the command makes me nervous. What if the new nick is in use, or is invalid, and the server rejects it? I'd rather update `self._nick` later, upon receiving the `NICK` event confirming the change back from the network.

But I also have to express frustration that during initial registration, the IRC protocol gives only silence—the absence of an error—as indication that the chosen NICK was accepted. This leap of faith seems unavoidable. 🙁